src/graphics/vertex-format.js

```diff
@@ -80,15 +132,19 @@ Object.assign(pc, function () {
         for (i = 0, len = description.length; i < len; i++) {
             var elementDesc = description[i];
             element = {
-                name: elementDesc.semantic,
+                semantic: elementDesc.semantic,
                 offset: 0,
                 stride: 0,
                 stream: -1,
                 scopeId: graphicsDevice.scope.resolve(elementDesc.semantic),
-                dataType: elementDesc.type,
-                numComponents: elementDesc.components,
+                type: elementDesc.type,
+                components: elementDesc.components,
                 normalize: (elementDesc.normalize === undefined) ? false : elementDesc.normalize,
-                size: elementDesc.components * _typeSize[elementDesc.type]
+                size: elementDesc.components * _typeSize[elementDesc.type],
+
+                name: elementDesc.semantic, // backwards compatibility
+                dataType: elementDesc.type, // backwards compatibility
+                numComponents: elementDesc.components // backwards compatibility
             };
```

Comment:
@willeastcott I appreciate your comment. But please note that the PR still has full backwards-compatibility. Neither the editor nor any app should be affected. The code changes I've made is basically introducing new properties (`semantic`, `type`, `components`) in the `pc.VertexFormat#elements` object to increase consistency and avoid duplication, but at the same time keep the old ones (`name`, `dataType`, `numComponents`) for backwards-compatibility.

All other code changes are just updating references to use the new properties, since they now have documentation. The old properties can still be used and reflect the exact same data. If you look through all references, you will note that these properties are also read-only.

Reply:
Could make name, dataType and numComponents properties instead?

Reply:
@slimbuck Do you mean renaming the other way around?

I think the new properties (semantic, type, components) is shorter and cleaner in general. But sure, that is also possible. Renaming properties in the constructor parameter will likely affect a lot more application code out there since it’s officially documented and has TS declarations (unlike elements). And even thought the API will be backwards-compatible in run-time, more devs will probably be greeted with warnings.

Reply:
@slimbuck Do you mean renaming the other way around?

I think the new properties (semantic, type, components) is shorter and cleaner in general. But sure, that is also possible. Renaming properties in the constructor parameter will likely affect a lot more application code out there since it’s officially documented and has TS declarations (unlike elements). And even thought the API will be backwards-compatible in run-time, more devs will probably be greeted with warnings.